### PR TITLE
fix: Fix service deployment deletion

### PR DESCRIPTION
**Pull request by maciaszczykm**
Improved unit tests to cover deletion process and tested manually on kind cluster connected to plrl-dev-aws.

### CODE REVIEW

**Review by michaeljguarino — APPROVED** [review]
what were the edge cases this covers?

File: go/controller/internal/controller/servicedeployment_controller.go [hunk 1]
```diff
@@ -496,57 +497,59 @@ func (r *ServiceReconciler) addOwnerReferences(ctx context.Context, service *v1a
 	return nil
 }
 
-func (r *ServiceReconciler) handleDelete(ctx context.Context, service *v1alpha1.ServiceDeployment) (ctrl.Result, error) {
-	log := log.FromContext(ctx)
-	if controllerutil.ContainsFinalizer(service, ServiceFinalizer) {
-		log.Info("try to delete service")
-		cluster := &v1alpha1.Cluster{}
```
[the remark this answers is not in the document]
Comment (michaeljguarino) [reply go/controller/internal/controller/servicedeployment_controller.go:503]: ah gotcha

File: go/controller/internal/controller/servicedeployment_controller.go [hunk 2]
```diff
@@ -143,7 +145,6 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 			utils.MarkCondition(service.SetCondition, v1alpha1.SynchronizedConditionType, v1.ConditionFalse, v1alpha1.SynchronizedConditionReasonError, err.Error())
 			return ctrl.Result{}, err
 		}
-		controllerutil.AddFinalizer(service, ServiceFinalizer)
```
Comment (maciaszczykm) [inline go/controller/internal/controller/servicedeployment_controller.go:146]: I got distracted and haven't finished second comment. There is also slight chance that finalizer was not added after resource creation. Now it's added before resource is created which should also make it more error proof.

File: go/controller/internal/controller/servicedeployment_controller.go [hunk 3]
```diff
@@ -496,57 +497,59 @@ func (r *ServiceReconciler) addOwnerReferences(ctx context.Context, service *v1a
...
+
+		// Stop reconciliation as the item does no longer exist.
+		return &ctrl.Result{}
 	}
-	return ctrl.Result{}, nil
+
+	return nil
 }
 
-func (r *ServiceReconciler) deleteService(service *v1alpha1.ServiceDeployment) error {
```
Comment (maciaszczykm) [inline go/controller/internal/controller/servicedeployment_controller.go:542]: Deletion was previously based on ID from CRD, but since we get object from API it's better to use it as it will always be there.
